lib/util/log.js

```diff
@@ -0,0 +1,8 @@
+'use strict';
+
+// eslint-disable-next-line node/no-missing-require
+const { createConsola } = require('consola/basic');
+
+const log = createConsola({ stdout: process.stderr });
```

Comment:
why put all output to `stderr`? that seems surprising. if we keep it we should at least add a comment to explain

Reply:
I did this to maintain the default behavior of `npmlog`. See [here ](https://github.com/npm/npmlog/blob/756bd05d01e7e4841fba25204d6b85dfcffeba3c/lib/log.js#L12)and the [README](https://github.com/npm/npmlog/blob/756bd05d01e7e4841fba25204d6b85dfcffeba3c/README.md)

Not exactly sure why this was the default 

Probably a good call to add a comment at least. Maybe `stderr` makes sense here since the logger outputs more debugging type output and `console.log()` is used for user-facing output?

Reply:
I think we should remove this rather than trying to maintain the behavior of `npmlog`. I checked (see below) and `consola` logs warnings and errors to stderr and info type logs to stdout, which seems like what I'd want and expect out of a logging system. Let's just check that we're setting the log levels appropriately whenever we do logging - I left one comment where I think we should do `log.error` to make sure an error goes to stdout

https://github.com/unjs/consola/blob/713ec3454889bf967e9c694bf800955f1cd60e52/src/reporters/basic.ts#L70
https://github.com/unjs/consola/blob/713ec3454889bf967e9c694bf800955f1cd60e52/src/constants.ts#L23

Reply:
Sure thing. Updated.

My thinking was to prevent potentially breaking dependents which rely on the `stderr` output for whatever reason but I agree that not using `stderr` for everything makes way more sense

Reply:
The main reason that most logging from `npm` and `node-gyp` goes to stderr (even now that neither use `npmlog`) is that stdout is reserved for output that is expected to be used programmatically (usually json). We don't usually swap which output stream is used based on flags like `--json` so the default is to place all logging on stderr so that if a json command is piped to something like `jq` the logging won't inadvertently create invalid json.

There are a lot of gray areas to this rule and this is something I spent a lot of time figuring out for `npm` in the past. I haven't actually review this code (yet) so I'm not saying that all output should be put on stderr, just that as a default stderr is usually better than stdout for CLI logging output.

Reply:
Thanks for this context @lukekarrys. Thoughts @benmccann?

Reply:
I have no objection to `stderr` given that additional context. If @lukekarrys thinks it's the way to go I'll defer to him on that

Reply:
@br0p0p do you want to change this back to use `stderr` by default? It might be the safer option to avoid changing that

Reply:
@benmccann sure, no problem